cloudfront/iiif/edge-lambda/src/dlcs_path_rewrite.ts

```diff
@@ -0,0 +1,11 @@
+import {
+    CloudFrontRequestHandler,
+} from 'aws-lambda';
+
+export const request: CloudFrontRequestHandler = (event, context, callback) => {
+    const request = event.Records[0].cf.request;
+
+    request.uri = request.uri.replace(/^\/image\/([A-Za-z0-9]+)(?:\.[Jj][Pp][Gg])?(\/.*)/gm, '/$1$2')
```

Comment:
Can you provide a comment explaining how this regex works? What bit of the original URL is it meant to capture and replace? My initial read of it was wrong, I think – I thought the `[Jj][Pp][Gg]` was on the end of `default.jpg`, not the image ID.

Reply:
Good idea. This is on the verge of "regex too complicated, simple thing instead".

Reply:
Maybe something like:

```
function convertLorisPathToDlcsPath(path: string) {
    const pathParts: Array<string> = path.split(",", 2)

    if (pathParts.length === 3 || pathParts[0] === "image") {
        const lorisImageId = pathParts[1]
        const dlcsImageId = lorisImageId.replace(/\.[Jj][Pp][Gg]/, "")
        path.replace(lorisImageId, dlcsImageId)
    } else {
        path
    }
} 
```